src/compiler.ts

```diff
@@ -3497,6 +3497,25 @@ export class Compiler extends DiagnosticEmitter {
     // not dealing with references from here on
     assert(!fromType.isReference && !toType.isReference);
 
+    // Early return if we have same types
+    if (toType.kind == fromType.kind) {
+      this.currentType = toType;
+      return expr;
+    }
+
+    // v128 to any / any to v128
+    // except v128 to bool
+    if (
+      toType.kind != TypeKind.BOOL &&
+      (toType.kind == TypeKind.V128 || fromType.kind == TypeKind.V128)
+    ) {
+      this.error(
+        DiagnosticCode.Type_0_is_not_assignable_to_type_1,
+        reportNode.range, fromType.toString(), toType.toString()
+      );
+      return module.unreachable();
+    }
```

Comment:
Can this perhaps be fused into the checks below that already do `isIntegerValue` etc.? `v128` is (and if if isn't yet should be) `isVectorValue` or so.

Reply:
It could be fused, but it just add a lot of noise. For every case we will need to add `this.error`. While we know that's V128 <-> any disallow for all cases except V128 -> bool. This signifintally simplify everything

Reply:
I would expect, though, that there'll one day more vector types than just `v128`, and that they all behave about the same, respectively cannot be assigned to each other. So perhaps covering this in a more generic way makes sense?

Reply:
Ah, I see. So you suggesting jsut switch `toType.kind == TypeKind.V128 || fromType.kind == TypeKind.V128` to `toType.isVectorValue || fromType.isVectorValue`?

Reply:
That'd be the first step, yeah, and integrating it into the chain of checks below would allow us to handle more common cases first I think, and more easily extend in case conversions between vectors would eventually be possible?

Reply:
> and more easily extend in case conversions between vectors would eventually be possible?

I'm not sure, as long as it creates more inconvenience and potentially errors. When and if such a moment comes, then it will be possible to reconsider the strategy to a more individualistic one.